### PR TITLE
[Messenger] [Redis] Remove BC layer around `delete_after_ack` option

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php [hunk 1]
```diff
@@ -50,13 +50,6 @@ public function __construct(?string $globalFailureReceiverName, ServiceProviderI
         parent::__construct();
     }
 
-    protected function getReceiverName(): string
-    {
-        trigger_deprecation('symfony/messenger', '5.3', 'The method "%s()" is deprecated, use getGlobalFailureReceiverName() instead.', __METHOD__);
-
-        return $this->globalFailureReceiverName;
-    }
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php:58]: unrelated change, but makes the bridge deprec-free. The class is internal so no need for a changelog entry (we didn't even add one when deprecating)

**Comment by chalasr** [comment]
fabbot is wrong (still struggles with union types) - PR ready

**Review by Nyholm — APPROVED** [review]
Thank you

**Review by OskarStark — APPROVED** [review]

**Comment by Nyholm** [comment]
Great work!

**Pull request merged**